rtf_vis_tool/src/tests/ViewFrustum.test.js

```diff
@@ -0,0 +1,110 @@
+        const img_w = 4;
+        const img_h = 4;
+        const fx = 5;
+        const ray_len = 0.5;
+
+        const testFrustum = new ViewFrustum(fx, img_w, img_h, ray_len);
+
+        const ray_dirs = testFrustum.compute_pixel_ray_directions_vectorized(uv, img_w, img_h);
+
+        // Ray dirs not normalized.
+        var expected_ray_dirs = nj.array([[-2,-2,5],
+                                          [-1,-1,5],
+                                          [0,0,5],
+                                          [1,1,5],
+                                          [2,2,5]]);
+        
+        // Normalizing ray dirs.
+        for (var row = 0; row < expected_ray_dirs.shape[0]; row++) {
+            var vector_mag = Math.sqrt(Math.pow(expected_ray_dirs.get(row,0),2) + 
+                                        Math.pow(expected_ray_dirs.get(row,1),2) + 
+                                        Math.pow(expected_ray_dirs.get(row,2),2));
+            for (var col = 0; col < ray_dirs.shape[1]; col++) {
+                expected_ray_dirs.set(row, col, (expected_ray_dirs.get(row,col)/vector_mag));
+            }
+        }
+        expect(expected_ray_dirs).toEqual(ray_dirs);
+    });
+
+    /* Initializes a test ViewFrustum object and checks if the outputted vertices in the camera coordinate
+       frame are correct.
+     */
+    it ('Test get_frustum_vertices_camfr.', () => {
+        const img_w = 4;
+        const img_h = 4;
+        const fx = 5;
+        const ray_len = 0.5;
+
+        const testFrustum = new ViewFrustum(fx, img_w, img_h, ray_len);
+        const vertices = testFrustum.get_frustum_vertices_camfr(); 
+        
+        const expected_vertices = [nj.array([0,0,0]),
+                                   nj.array([(-1/Math.sqrt(33)), (-1/Math.sqrt(33)), (5/(2*Math.sqrt(33)))]),
+                                   nj.array([(1/(2*Math.sqrt(30))), (-1/Math.sqrt(30)), (5/(2*Math.sqrt(30)))]),
+                                   nj.array([(1/(2*Math.sqrt(27))), (1/(2*Math.sqrt(27))), (5/(2*Math.sqrt(27)))]),
+                                   nj.array([(-1/Math.sqrt(30)), (1/(2*Math.sqrt(30))), (5/(2*Math.sqrt(30)))])];
+        
+        expect(expected_vertices).toEqual(vertices);
+    });
+
+    /* Creates some sample ray directions and checks if they are scaled properly according to the scalar: ray_len. */
+    it ('Test calculate_scaled_ray_dirs.', () => {
+        const img_w = 4;
+        const img_h = 4;
+        const fx = 5;
+        const ray_len = 0.5;
+
+        const testFrustum = new ViewFrustum(fx, img_w, img_h, ray_len);
+        const ray_dirs = nj.array([[1,1,1],
+                                  [2,2,2],
+                                  [3,3,3],
+                                  [4,4,4],
+                                  [5,5,5]]);
+        const scaled_ray_dirs = testFrustum.calculate_scaled_ray_dirs(ray_dirs, ray_len);
+        const expected_scaled_ray_dirs = [nj.array([0,0,0]), 
```

Comment:
should this be 0.5,0.5,0.5 not 0,0,0?

Reply:
I don't think so, because in the `calculate_scaled_ray_dirs` function in `view_frustum.js`, `v0` is just set to the origin point `[0,0,0]` when all 5 vertex vectors are scaled.